pilot/pkg/dataplane/interfaces.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataplane
+
+import (
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+// Dataplane represents the interfaces to be implemented by code that generates xDS responses
+type Dataplane interface {
```

Comment:
I don't think interface will scale. It needs to be more fine grained, so we can re-compute the diffs only.

Reply:
Thanks, but this is a start. I do not want to boil the ocean and define differential computation upfront. We first need to move to v1alpha3+v2. Once the code takes shape, we can optimize this.

Reply:
What's the purpose of this interface then? Is it to unify certain implementations? Is it to set the goal for the implementation?